pkg/config/config.go

```diff
@@ -24,6 +24,7 @@ type Build struct {
 	PythonFindLinks      []string `json:"python_find_links,omitempty" yaml:"python_find_links"`
 	PythonPackages       []string `json:"python_packages,omitempty" yaml:"python_packages"`
 	Run                  []string `json:"run,omitempty" yaml:"run"`
+	BuildEnv             []string `json:"env,omitempty" yaml:"env"`
```

Comment:
This should be `environment` for consistency with Docker Compose.

It also probably doesn't need the `Build` prefix because it's already in a struct called `Build`. So maybe `Environment` if we're also renaming to "environment"?

Reply:
<details><summary> For other reviewers /  posterity, here's the discussion about this from the previous PR</summary>

![image](https://user-images.githubusercontent.com/2420688/151422914-fa0e4345-5c12-4c7a-96ed-b7c92ed1c8e5.png)

(From PR #361 )

</details>

Reply:
At first I felt differently about this, because `environment` in Docker Compose has its own semantics, but I've come around to agreeing that `build.environment` makes the most sense!

Why I agree now:

* `environment` looks plenty clear in `cog.yaml`; it's under `build`
*  although `env` might resemble `ENV` to me, I think it's because of handwriting lots of Dockerfiles.. but really, `env` is not `ENV`. If someone's experienced with Python (and/or Docker Compose) but not Dockerfiles, `env` might resemble/remind of `virtualenv` or `pyenv`, and that would be confusing.

Reply:
[It's resolved, but leaving it open for now for easier review.]

Reply:
@bfirsh I also revised the Pull Request description so that the usage example is right (`build.environment` not `build.env`)